### PR TITLE
🧪 Add missing error edge case tests for parseApiErrorMessage

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The testing gap addressed was missing tests for error text that is valid JSON but throws TypeError when checking `.error` (e.g. `null`, array, number) or typical edge cases like empty string and HTML payload.
📊 **Coverage:** Added coverage for `null`, arrays, empty strings, and HTML error pages to handle various scenarios inside `parseApiErrorMessage`.
✨ **Result:** Enhanced the resilience and reliability of the api error handling test suite.

---
*PR created automatically by Jules for task [4948008687202997452](https://jules.google.com/task/4948008687202997452) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `parseApiErrorMessage` に対して、`null`・配列・空文字・HTMLレスポンスという4つのエッジケーステストを追加します。追加されたテストの期待値はいずれも実装と一致しており、テストとして正しく機能します。
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

テストのみの変更であり、プロダクションコードへの影響はなく、マージは安全です。

追加された4つのテストはいずれも実装の挙動と一致しており、正しく動作します。ただし、新規追加の `null` および空文字ケースで `defaultMessage` を渡したときの挙動がカバーされておらず、カバレッジに若干の抜けがあります。また配列ケースのテスト説明が実際の挙動（TypeError ではなく `undefined` の素通り）と一致していません。

特に注意が必要なファイルはありませんが、`api.test.ts` の新規テストケースに `defaultMessage` 組み合わせのテストを補完すると、より堅牢なテストスイートになります。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web-app-vercel/lib/__tests__/api.test.ts | `parseApiErrorMessage` のエッジケーステスト4件を追加。各テストの動作予測は実装と一致しているが、新規追加ケース（`'null'`・空文字）に `defaultMessage` を渡したケースのカバレッジが不足している。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["parseApiErrorMessage(errorText, defaultMessage?)"] --> B["JSON.parse(errorText)"]
    B -->|SyntaxError| C["catch ブロック"]
    B -->|"null を返す"| D["null.error → TypeError"]
    B -->|"配列を返す"| E[".error → undefined falsy"]
    B -->|"オブジェクトを返す"| F{"errorJson.error ?"}
    D --> C
    C --> G["return defaultMessage OR errorText"]
    E --> G
    F -->|truthy| H["return errorJson.error"]
    F -->|falsy| G
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
packages/web-app-vercel/lib/__tests__/api.test.ts:56-58
**配列ケースの TypeError 説明が不正確**

PR説明では配列(`[]`)を「`.error` 参照時に TypeError を投げる」ケースとして挙げていますが、実装上 `[].error` は `undefined`（TypeError なし）を返し、単に `if` の条件が偽になるだけです。テスト自体の期待値は正しいですが、`null` ケース（本当に TypeError が発生する）と混同されており、今後のメンテナンス時に混乱を招く可能性があります。テスト名やコメントで「`.error` が `undefined` のため素通り」と明記することを検討してください。

### Issue 2 of 2
packages/web-app-vercel/lib/__tests__/api.test.ts:60-62
**空文字 × `defaultMessage` 組み合わせが未テスト**

`parseApiErrorMessage('', 'fallback')` を呼ぶと、JSON.parse が SyntaxError を投げて catch に入り、`return 'fallback' || ''` = `'fallback'` が返ります。同様に `parseApiErrorMessage('null', 'fallback')` も TypeError が catch されて `'fallback'` を返します。現行テストはいずれも `defaultMessage` なし呼び出しのみで、`defaultMessage` が指定されたときの挙動がカバーされていません。既存の `returns default message when provided and JSON has no error field` ケースのように対称テストを追加することを検討してください。

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test: add missing tests for parseApiErro..."](https://github.com/hiroki-org/audicle/commit/0bafe992d2d4dafe1e7dc8c6a1cd32886d878f15) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30793012)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: packages/web-app-vercel/lib/__tests__/api.test.ts [hunk 1]
```diff
@@ -48,6 +48,24 @@ describe('parseApiErrorMessage', () => {
...
         expect(parseApiErrorMessage(errorText, 'デフォルトメッセージ')).toBe('エラーメッセージ');
     });
+
+    it('returns original text when errorText is "null" (valid JSON but not an object)', () => {
+        expect(parseApiErrorMessage('null')).toBe('null');
+    });
+
+    it('returns original text when errorText is an array "[]"', () => {
+        expect(parseApiErrorMessage('[]')).toBe('[]');
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/api.test.ts:58]: 対応しました。配列ケースのテスト名を、TypeError ではなく error field が undefined になる挙動として明確化しました。

File: packages/web-app-vercel/lib/__tests__/api.test.ts [hunk 2]
```diff
@@ -48,6 +48,24 @@ describe('parseApiErrorMessage', () => {
         const errorText = JSON.stringify({ error: 'エラーメッセージ' });
         expect(parseApiErrorMessage(errorText, 'デフォルトメッセージ')).toBe('エラーメッセージ');
     });
+
+    it('returns original text when errorText is "null" (valid JSON but not an object)', () => {
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/api.test.ts:52]: 対応しました。null のテスト名を「valid JSON but null」に修正し、実際に TypeError 経由で fallback される値だと分かる表現にしました。

File: packages/web-app-vercel/lib/__tests__/api.test.ts [hunk 3]
```diff
@@ -48,6 +48,24 @@ describe('parseApiErrorMessage', () => {
...
+    });
+
+    it('returns original text when errorText is an empty string', () => {
+        expect(parseApiErrorMessage('')).toBe('');
+    });
+
+    it('returns original text when errorText is an HTML error page', () => {
+        const htmlError = '<html><body>502 Bad Gateway</body></html>';
+        expect(parseApiErrorMessage(htmlError)).toBe(htmlError);
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/api.test.ts:67]: 対応しました。number の 123 ケースと、null + defaultMessage の fallback ケースを追加しました。HTML error page の既存ケースも維持しています。

File: packages/web-app-vercel/lib/__tests__/api.test.ts [hunk 4]
```diff
@@ -48,6 +48,24 @@ describe('parseApiErrorMessage', () => {
...
+        expect(parseApiErrorMessage('null')).toBe('null');
+    });
+
+    it('returns original text when errorText is an array "[]"', () => {
+        expect(parseApiErrorMessage('[]')).toBe('[]');
+    });
+
+    it('returns original text when errorText is an empty string', () => {
+        expect(parseApiErrorMessage('')).toBe('');
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/api.test.ts:62]: 対応しました。空文字 + defaultMessage と null + defaultMessage の対称テストを追加し、catch 経由で defaultMessage が返る挙動を明示しました。
